src/nfa/nfa.cc

```diff
@@ -429,26 +429,77 @@ bool Nfa::is_flat() const {
     return flat;
 }
 
-std::string Nfa::print_to_dot(const bool ascii) const {
+std::string Nfa::print_to_dot(const bool decode_ascii_chars, const bool use_intervals, const int max_label_length) const {
     std::stringstream output;
-    print_to_dot(output, ascii);
+    print_to_dot(output, decode_ascii_chars, use_intervals, max_label_length);
     return output.str();
 }
 
-void Nfa::print_to_dot(std::ostream &output, const bool ascii) const {
-    auto to_ascii = [&](const Symbol symbol) {
+void Nfa::print_to_dot(std::ostream &output, const bool decode_ascii_chars, const bool use_intervals, const int max_label_length) const {
+    auto to_ascii = [&](const Symbol symbol) -> std::string {
         // Translate only printable ASCII characters.
-        if (symbol < 33) {
+        if (symbol < 33 || symbol >= 127) {
             return "<" + std::to_string(symbol) + ">";
         }
         switch (symbol) {
-            case '"':     return std::string("\\\"");
-            case '\\':    return std::string("\\\\");
-            case EPSILON: return std::string("<eps>");
+            case '"':     return "\\\"";
+            case '\\':    return "\\\\";
             default:      return std::string(1, static_cast<char>(symbol));
         }
     };
 
+    auto translate_symbol = [&](const Symbol symbol) -> std::string {
+        switch (symbol) {
+            case EPSILON:      return "<eps>";
+            default:           break;
+        }
```

Comment:
This can be simpler:
```suggestion
        if (symbol == EPSILON) {
            return "<eps>";
        }
```

Reply:
I know. I put the switch there on purpose, so when we introduce new — I don't know how many — symbols, it won't become a mess of if-else statements.